### PR TITLE
Enforce OSD option

### PR DESCRIPTION
**Pull request by haslinghuis**
Lots of issues reported with users not reading release notes so decided to enforce OSD option.

image

### CODE REVIEW

**Comment by VitroidFPV** [comment]
For the wording: as this is meant to be a beginner-accessible feature (most people have no clue what a "MAX7456" is), maybe just Digital and Analog? Other than that, looks great

**Review by VitroidFPV — APPROVED** [review]

**Review by VitroidFPV — APPROVED** [review]

File: src/js/tabs/firmware_flasher.js [hunk 1]
```diff
@@ -220,8 +220,26 @@ firmware_flasher.initialize = function (callback) {
...
+
+            // pre select osd option when found in general options after auto detect
+            if (self.cloudBuildOptions) {
+                data.osdProtocols.forEach((option) => {
+                    option.default = self.cloudBuildOptions.includes(option.value);
+                });
+            }
+
+            // remove osdProtocols from generalOptions
+            data.generalOptions = data.generalOptions.filter(option => option.value !== 'USE_FRSKYOSD' && option.value !== 'USE_OSD_SD' && option.value !== 'USE_OSD_HD');
```
Comment (blckmn) [inline src/js/tabs/firmware_flasher.js:238]:
```suggestion
            data.generalOptions = data.generalOptions.filter(option => !option.group);
```

File: src/js/tabs/firmware_flasher.js [hunk 2]
```diff
@@ -220,8 +220,26 @@ firmware_flasher.initialize = function (callback) {
                 return;
             }
 
+            data.osdProtocols = [
```
Comment (blckmn) [inline src/js/tabs/firmware_flasher.js:223]:
```suggestion
            data.osdProtocols = data.generalOptions.filter(option => option.group === 'OSD');
```

Comment (blckmn) [reply src/js/tabs/firmware_flasher.js:223]: You can map to a new item, using the groupedName if you want to change the name to a different name.

**Review by blckmn — CHANGES_REQUESTED** [review]

File: src/js/tabs/firmware_flasher.js (hunk 2 above)
Comment (blckmn) [reply src/js/tabs/firmware_flasher.js:223]: https://build.betaflight.com/api/options

Comment (blckmn) [reply src/js/tabs/firmware_flasher.js:223]: you will also need to cater for the "none" option... so maybe a helper function to add this, and map. note the none option should have an empty value.

**Comment by nerdCopter** [comment]
`8dd138ce`
image

**Comment by haslinghuis** [comment]
@nerdCopter do you have more info on what kind of build? Works with cloud build target

**Comment by nerdCopter** [comment]
`0f54217e` works
image

word `Analogue` is incorrect, should be `Analog`

opinion: I feel that `Analog` or `Digital` should be default, not `None`.

**Comment by haslinghuis** [comment]
@nerdCopter 

> word Analogue is incorrect, should be Analog

That is how it is named in cloud build API.

> opinion: I feel that Analog or Digital should be default, not None.

None is only selected if analog, digital, frsky is not present.

**Comment by nerdCopter** [comment]
> That is how it is named in cloud build API.

https://www.merriam-webster.com/dictionary/analogue  versus https://www.merriam-webster.com/dictionary/analog.  okay, i thin it is British English vs American English issue.

> None is only selected if analog, digital, frsky is not present.

fair, i only saw loading screen , no FC.

**Comment by blckmn** [comment]
> > That is how it is named in cloud build API.
> 
> https://www.merriam-webster.com/dictionary/analogue versus https://www.merriam-webster.com/dictionary/analog. i think it is British English vs American English issue.
> 
> > None is only selected if analog, digital, frsky is not present.
> 
> fair, i only saw loading screen , no FC.

I have updated this to be "american". It will flow through in an hour or so. 

I does however raise a good point. We should localise these display values. We can do so by adding the american version into the messages.json as a key, and then adding the localisation after request to API. This would allow for crowdin to do the work. However we could also consider adding the localisation to the API based on the language setting provided in the header.

**Review by blckmn — APPROVED** [review]

**Review by nerdCopter — APPROVED** [review]

**Pull request merged**